### DISCUSSION TITLE
Deserialize js message and ack

### DISCUSSION THREAD

**Discussion by smiziara** [nats-io/nats.deno#328] [Q&A] [answered]
Hello Guys,

We have a scenario here that we receive Jetstream messages, and have to serialize them. Later we need to deserialize and Ack or Nack the message.
How we can rebuild the object so we can ack that message?

Thanks!

**Comment by aricart** [comment, marked as answer]
there are references to the client (for respond, etc) in the message, so not sure that serializing the actual js message is a good idea. Instead, you should be reading the payload/and headers if needed, and storing those (as that is your data).

If you want to ack the message, you can try (you may need to add a ):
```typescript
// @ts-ignore: accessing internals
const nc // your nats connection
const reply = m.reply;
const sc = StringCodec();
nc.publish(reply, sc.encode("+ACK"));
// or 
nc.publish(reply, sc.encode("-NAK");
```

Note that if your operation takes longer than the amount configured to wait for acks, another client could have processed or reprocessed your message.

Not sure why the service that receives the message cannot await a response from your downstream or simply process your payload, and simply ack.

**Reply by smiziara** [reply]
@aricart We are not actually serializing the message object from the SDK, what we need is, from a previous serialized information, send a NAK/ACK to it, and since the object instances are lost, we were not sure.
This system is a declarative workflow so when passing from one node to the next, we might lose the object instance, we can only guarantee the data. 

So basically we need to store the payload, headers and the reply object, that will be used to generate the ACK/NAK topics for Jetstream, like you sent?

**Reply by aricart** [reply]
Yes, all you should have to do then is copy the reply subject from the message, and then publish to that subject with the payload indicating ack/or nak as shown above.
